### PR TITLE
feat(NavBarLink)!: renamed to NavBarButton; improved styles

### PR DESCRIPTION
**Pull request by slingner**
Related to issue #74

NavbarLink is now NavBarButton.
Made a new story specifically for NavBarButton (took out of navbar story), added variations for all pseudo States.

### CODE REVIEW

**Review by WilliamKelley — CHANGES_REQUESTED** [review]
Please validate all styles with Figma again. Icon size doesn't match:

PR:
image

Figma:
image

File: libs/spark/src/NavBarButton.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          fill: theme.palette.blue[3],
+          fillOpacity: '0.24',
+        },
+      },
+    },
+    label: { fontWeight: 700 },
+  },
+  iconSizeLarge: {
+    '& > *:first-child': {
+      fontSize: '2rem', // small/medium/large for icons is 16px/24px/32px
```
Comment (WilliamKelley) [inline libs/spark/src/NavBarButton.tsx:49]:
```suggestion
      fontSize: '2rem',
```

File: libs/spark/stories/navbarbutton.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        Link
+      </NavBarButton>
+    </span>
+  </GridContainer>
+);
+
+export const All = AllTemplate.bind({});
+export const AllDisabled = AllTemplate.bind({});
+AllDisabled.args = { disabled: true };
+export const AllHover = AllTemplate.bind({});
```
Comment (WilliamKelley) [inline libs/spark/stories/navbarbutton.stories.tsx:95]:
```suggestion
export const All = AllTemplate.bind({});

export const AllDisabled = AllTemplate.bind({});
AllDisabled.args = { disabled: true };

export const AllHover = AllTemplate.bind({});
```

etc. for readability

File: libs/spark/test/navbar.spec.tsx [hunk 3]
```diff
@@ -1,17 +1,16 @@
 import { render } from '@testing-library/react';
 import 'jest-styled-components';
-import { SparkThemeProvider, NavBar, NavbarLink } from '../src';
-import QuestionDuotoneIcon from '@prenda/spark-icons/QuestionDuotone';
+import { SparkThemeProvider, NavBar, NavBarButton } from '../src';
+import QuestionDuotone from '@prenda/spark-icons/QuestionDuotone';
 
 describe('Button', () => {
```
Comment (WilliamKelley) [inline libs/spark/test/navbar.spec.tsx:6]: Not this PR's fault, but it can fix it

```suggestion
describe('NavBar', () => {
```

**Review by WilliamKelley — CHANGES_REQUESTED** [review]
Looks great! just little things.

I'll also suggest two things related to the NavBar story that isn't the responsibility of this PR, but you can choose to fix them:
- use correct new icons for the nav bar buttons
- match the notifications button styling in figma (less space between the icon and '0' -- will require a one-off style customization.

**Comment by slingner** [comment]
> Looks great! just little things.
> 
> I'll also suggest two things related to the NavBar story that isn't the responsibility of this PR, but you can choose to fix them:
> 
> * use correct new icons for the nav bar buttons
> * match the notifications button styling in figma (less space between the icon and '0' -- will require a one-off style customization.

I ended up added a negative margin to get the 0 closer to the icon, let me know if you have a cleaner solution in mind!

**Comment by WilliamKelley** [comment]
> I ended up added a negative margin to get the 0 closer to the icon, let me know if you have a cleaner solution in mind!

That does the job! But I think we should set the example to not use inline styles. Instead,
```jsx

import { withStyles } from '@material-ui/core';

const CustomToolbar = withStyles({
  root: {
    gap: 8,
  },
})(Toolbar);

const InboxNavBarButton = withStyles({
  root: {
    marginLeft: 'auto',
  },
  startIcon: {
    marginRight: 2,
  },
})(NavBarButton);
```

File: libs/spark/stories/navbar.stories.tsx [hunk 4]
```diff
@@ -48,19 +64,17 @@ const Template = (args: NavBarProps) => (
...
         Create
       </NavBarButton>
-      <NavBarButton
+      <InboxNavBarButton
         href="#"
-        startIcon={
-          <InboxFilledDuotone style={{ marginRight: -6 }} fontSize="large" />
-        }
+        startIcon={<InboxFilledDuotone fontSize="large" />}
         style={{ marginLeft: 'auto' }}
```
Comment (WilliamKelley) [inline libs/spark/stories/navbar.stories.tsx:70]:
```suggestion
```

**Review by WilliamKelley — CHANGES_REQUESTED** [review]

**Review by WilliamKelley — APPROVED** [review]

**Review by shadoath — APPROVED** [review]

**Pull request merged**